### PR TITLE
Add `pathvar` command

### PR DESCRIPTION
**Pull request by nathom**
This PR implements the `pathvar` command, which allows for temporary modifications to the `PATH` variable. This is necessary for virtualenv support (#852), among other things.

**Demo**
```
/Users/nathan/nushell(main)> pathvar
───┬─────────────────────────────────────────
 0 │ /users/nathan/.pyenv/shims
 1 │ /users/nathan/.local/bin
 2 │ /Users/nathan/.cargo/bin
 3 │ /Users/nathan/.pyenv/versions/3.9.5/bin
 4 │ /usr/local/bin
 5 │ /usr/bin
 6 │ /bin
 7 │ /usr/sbin
 8 │ /sbin
───┴─────────────────────────────────────────

/Users/nathan/nushell(main)> pathvar add /my/directory
/Users/nathan/nushell(main)> pathvar
───┬─────────────────────────────────────────
 0 │ /my/directory
 1 │ /users/nathan/.pyenv/shims
 2 │ /users/nathan/.local/bin
 3 │ /Users/nathan/.cargo/bin
 4 │ /Users/nathan/.pyenv/versions/3.9.5/bin
 5 │ /usr/local/bin
 6 │ /usr/bin
 7 │ /bin
 8 │ /usr/sbin
 9 │ /sbin
───┴─────────────────────────────────────────

/Users/nathan/nushell(main)> pathvar remove 0
/Users/nathan/nushell(main)> pathvar
───┬─────────────────────────────────────────
 0 │ /users/nathan/.pyenv/shims
 1 │ /users/nathan/.local/bin
 2 │ /Users/nathan/.cargo/bin
 3 │ /Users/nathan/.pyenv/versions/3.9.5/bin
 4 │ /usr/local/bin
 5 │ /usr/bin
 6 │ /bin
 7 │ /usr/sbin
 8 │ /sbin
───┴─────────────────────────────────────────

/Users/nathan/nushell(main)> pathvar append /other/directory
/Users/nathan/nushell(main)> pathvar save
/Users/nathan/nushell(main)> config get path
───┬─────────────────────────────────────────
 0 │ /users/nathan/.pyenv/shims
 1 │ /users/nathan/.local/bin
 2 │ /Users/nathan/.cargo/bin
 3 │ /Users/nathan/.pyenv/versions/3.9.5/bin
 4 │ /usr/local/bin
 5 │ /usr/bin
 6 │ /bin
 7 │ /usr/sbin
 8 │ /sbin
 9 │ /other/directory
───┴─────────────────────────────────────────

/Users/nathan/nushell(main)>
```

### CODE REVIEW

File: crates/nu-command/src/commands/pathvar/append.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+pub fn add(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let ctx = &args.context;
+    let path_to_append_arg: Tagged<PathBuf> = args.req(0)?;
+    let path_to_append = path_to_append_arg.item.into_os_string().into_string();
+
+    if let Ok(path) = path_to_append {
+        let mut pathvar = ctx.scope.get_env(NATIVE_PATH_ENV_VAR).unwrap();
+        pathvar.push(':');
```
Comment (fdncred) [inline crates/nu-command/src/commands/pathvar/append.rs:44]: Won't this break Windows functionality?

You may need something like this, that exists in the `char` command. Or prehaps relocate it to somewhere more common that everyone can use.

```rust
// Character used to separate directories in a Path Environment variable on windows is ";"
#[cfg(target_family = "windows")]
const ENV_PATH_SEPARATOR_CHAR: char = ';';
// Character used to separate directories in a Path Environment variable on linux/mac/unix is ":"
#[cfg(not(target_family = "windows"))]
const ENV_PATH_SEPARATOR_CHAR: char = ':';
```

File: crates/nu-command/src/commands/pathvar/command.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+}
+
+pub fn get_pathvar(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let pathvar: Vec<Value> = args
+        .context
+        .scope
+        .get_env(NATIVE_PATH_ENV_VAR)
+        .unwrap()
+        .split(':')
```
Comment (fdncred) [inline crates/nu-command/src/commands/pathvar/command.rs:53]: Same comment as the other. `:` probably beaks the functionality on windows.

File: crates/nu-command/src/commands/pathvar/add.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+pub fn add(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let ctx = &args.context;
+
+    let path_to_add: Tagged<PathBuf> = args.req(0)?;
+    let path = path_to_add.item.into_os_string().into_string();
+
+    if let Ok(mut path) = path {
+        path.push(':');
```
Comment (fdncred) [inline crates/nu-command/src/commands/pathvar/add.rs:44]: `:` = not good on windows

File: crates/nu-command/src/commands/pathvar/remove.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+}
+
+pub fn remove(args: CommandArgs) -> Result<OutputStream, ShellError> {
+    let ctx = &args.context;
+    let index_to_remove_arg: Tagged<u64> = args.req(0)?;
+    let index_to_remove = index_to_remove_arg.item as usize;
+
+    let old_pathvar = ctx.scope.get_env(NATIVE_PATH_ENV_VAR).unwrap();
+    let mut paths: Vec<&str> = old_pathvar.split(':').collect();
```
Comment (fdncred) [inline crates/nu-command/src/commands/pathvar/remove.rs:46]: `:` breaks on windows

File: crates/nu-command/src/commands/pathvar/reset.rs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        // let row = Value::new(
+        //     UntaggedValue::Table(paths),
+        //     Tag::from(Span::from(&span_range)),
+        // );
+
+        let default_pathvar = global_cfg.vars.get("path");
+        if let Some(pathvar) = default_pathvar {
+            trace!("default_pathvar: {:?}", pathvar);
+            if let UntaggedValue::Table(paths) = &pathvar.value {
+                let pathvar_str = paths.iter().map(|x| x.as_string().unwrap()).join(":");
```
Comment (fdncred) [inline crates/nu-command/src/commands/pathvar/reset.rs:45]: `:` breaks on windows

File: crates/nu-command/src/commands/pathvar/append.rs (hunk 1 above)
Comment (nathom) [reply crates/nu-command/src/commands/pathvar/append.rs:44]: Oh, I didn't know windows had a different separator. I'll fix that. Thanks.

**Comment by nathom** [comment]
I think it's good to go. Let me know if there are any other changes I need to make.

**Comment by jonathandturner** [comment]
@nathom - I had a few additional ideas, but I think this is good to land and continue evolving.

Something we recently added was `load-env` where you can take in a changeset to add to the environment in scope. I was wondering if there might be a way to do something similar with `pathvar`. Might be worth a brainstorm on an issue or in the discord.

**Pull request merged**